tests/ui/unsized/unsized3-rpass.rs

```diff
@@ -59,7 +59,7 @@ pub fn main() {
         }
 
         let data: Box<Foo_<i32>> = Box::new(Foo_ { f: [1, 2, 3] });
-        let x: &Foo<i32> = mem::transmute(slice::from_raw_parts(&*data, 3));
+        let x: &Foo<i32> = mem::transmute(ptr::slice_from_raw_parts(&*data, 3));
```

Comment:
I don't think I understand why this helps, given that both are transmuted to the same target type?

Reply:
Oh so the temporary slice actually pretends to have 3 `Foo_<i32>`. Wtf, why? At that point one might just transmute a `(ptr, usize)` directly without going through from_raw_parts...

Anyway, patch looks good.

Reply:
> transmute a `(ptr, usize)`

That would require relying on the still-unstable slice layout :wink: 

But yeah I have no idea why anyone would want to write the code in this test. Considering the amount of code in the SIMD tests that were using a pointer to the first element as a pointer to the whole array it might not hurt for someone to skim over the tests for their favorite feature...

Reply:
It's possible to make a DST ref without relying on `transmute` or that any two slice-like fat pointers have the same layout by using the fact that `as` preserves metadata.  [playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=bcaabbf0fec011c33f9338c2d078ca71)

Reply:
Yeah that's the sort of thing I would advise someone write, but this is a test... so I am not really sure what it is supposed to test.